src/d3a/models/strategy/load_hours.py

```diff
@@ -164,7 +164,7 @@ def _delete_past_state(self):
                 to_delete.append(k)
         for k in to_delete:
             del self.energy_requirement_Wh[k]
-            del self.state.desired_energy_Wh[k]
+            self.state.desired_energy_Wh.pop(k, None)
```

Comment:
weird that this is needed. There should always be a `desired_energy_Wh` once the state was initiated. Right?

Reply:
Should not be needed. Some of the commits here are before the hackathon where we fixed these issues while debugging Canary network, that's why these should not be valid anymore.

Reply:
AH, OK, I was browsing through the commits. Sorry.